inc/hclib-promise.h

```diff
@@ -73,31 +73,15 @@ typedef struct _hclib_future_t {
     struct hclib_promise_st *owner;
 } hclib_future_t;
 
-/**
- * An hclib_triggered_task_t associates dependent tasks and the promises that
- * trigger their execution. This is exposed so that the runtime knows the size
- * of the struct.
- */
-typedef struct hclib_triggered_task_st {
-    // NULL-terminated list of futures this task is registered on
-    hclib_future_t **waiting_frontier;
-    /*
-     * This allows us to chain all dependent tasks waiting on a same promise.
-     * Whenever a triggered task wants to register on a promise, and that
-     * promise is not ready, we chain the current triggered task and the
-     * promise's wait_list_head and try to cas on the promise's wait_list_head,
-     * with the current triggered task.
-     */
-    struct hclib_triggered_task_st *next_waiting_on_same_future;
-} hclib_triggered_task_t;
+struct hclib_task_t;
 
 // We define a typedef in this unit for convenience
 typedef struct hclib_promise_st {
     hclib_future_t future;
-	int kind;
-    volatile void *datum;
+    promise_kind_t kind;
+    void *volatile datum;
```

Comment:
I'm pretty sure the original author thought that `volatile void*` was the equivalent of a volatile reference in Java—but it's not. (The HC runtime was designed based on the HJ runtime, so it wouldn't surprise me if someone was just translating a Java class into a C struct and did this.)
Since these fields in the promise struct are what get modified concurrently (not the values they point to), I think the intent was to mark the pointers themselves `volatile`.
However, as I pointed out in #18, I think we should be able to rid our codebase of `volatile` altogether. Using `__sync`/`__atomic` builtins should already be providing the memory barriers that we need to avoid stale values, so I don't think `volatile` is doing anything useful...
I left these `volatile`s for now because I'm only *pretty sure* that we don't need them—but I'm 100% sure that they don't make any sense on the other side of the `*`.